numba/np/arraymath.py

```diff
@@ -4422,3 +4422,45 @@ def impl(a, b):
         return _cross2d_operation(a_, b_)
 
     return impl
+
+
+#----------------------------------------------------------------------------
+# Equality tests implemented on Floats
+# Implemented based on this feature request
+# - https://github.com/numba/numba/issues/5977
+@overload(np.isclose)
+def np_isclose(a, b, rtol=1e-05, atol=1e-08, equal_nan=False):
+    if not type_can_asarray(a) or not type_can_asarray(b):
+        raise TypingError("Inputs a and b must be array-like.")
+    if not isinstance(rtol, types.Float) or not isinstance(atol, types.Float):
+        raise TypingError("Relative and absolute tolerance must " +
+                          "be represented as floats.")
```

Comment:
I think they just need to be non-complex numbers? e.g. `np.isclose(1, 2, np.uint(3))` works fine. I think something like `if <the type> not in types.real_domain | types.integer_domain:` could work.